src/catalog/catalog_structs.cpp

```diff
@@ -69,6 +69,15 @@ Property TableSchema::getProperty(property_id_t propertyID) const {
         "Table: %s doesn't have a property with propertyID=%d.", tableName.c_str(), propertyID));
 }
 
+void TableSchema::renameProperty(property_id_t propertyID, string newName) {
+    for (auto& property : properties) {
+        if (property.propertyID == propertyID) {
+            property.name = newName;
+            return;
+        }
+    }
```

Comment:
let's through an InternalException here, i.e. we should always be able to find the given propertyID

Reply:
added a statement to throw exception